Mta 4707 (#3300)
scaleway/docs-content: tutorials/easydeploy-gitlab-server/index.mdx

---
meta:
  title: Deploy GitLab Server on Scaleway Kubernetes clusters using Easy Deploy

>   description: Learn how to deploy GitLab server on Scaleway Kubernetes clusters using the Easy Deploy feature. This comprehensive guide provides step-by-step deployment instructions, and post-deployment configurations necessary for optimal continuous delivery and GitOps practices.

Author: ```suggestion
  description: Deploy GitLab server on Scaleway Kubernetes clusters with Easy Deploy. This guide provides steps for deployment and configurations for optimal GitOps practices.
```

content:
  h1: Deploy GitLab Server on Scaleway Kubernetes clusters using Easy Deploy

>   paragraph: Learn how to deploy GitLab server on Scaleway Kubernetes clusters using the Easy Deploy feature. This comprehensive guide provides step-by-step deployment instructions, and post-deployment configurations necessary for optimal continuous delivery and GitOps practices.

Author: ```suggestion
  paragraph: Deploy GitLab server on Scaleway Kubernetes clusters with Easy Deploy. This guide provides steps for deployment and configurations for optimal GitOps practices
```

tags: GitLab server kubernetes easy deploy
categories:
  - containers
dates:
  validation: 2024-06-20
  posted: 2024-06-20
---

> This tutorial provides a guide to deploying a GitLab server on Scaleway Kubernetes clusters using the Easy Deploy feature.

Reviewer: ```suggestion
This tutorial provides a guide to deploy a GitLab server on Scaleway Kubernetes clusters using the Easy Deploy feature.
```

It outlines the prerequisites, step-by-step deployment instructions, and post-deployment configurations necessary for optimal continuous delivery and GitOps practices.

<Macro id="requirements" />
- A Scaleway account logged into the [console](https://console.scaleway.com)
- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- A valid [API key](/identity-and-access-management/iam/how-to/create-api-keys/)
- Created a Scaleway Kubernetes [Kapsule](/containers/kubernetes/how-to/create-cluster/) or [Kosmos](/containers/kubernetes/how-to/create-kosmos-cluster/) cluster

> - A domain name and access to the DNS zone of it

Reviewer: ```suggestion
- A domain name and access to its DNS zone
```

## Deploying the GitLab Server application using Easy Deploy

1. In the [Scaleway console](https://console.scaleway.com/), navigate to the **Kubernetes** section under **Containers**.
2. Select the cluster where you want to deploy the GitLab server. The **Cluster Information** tab will display.
3. Click the **Easy Deploy** tab to open the application dashboard.
4. Click **Deploy Application** to launch the deployment wizard.
5. Select **Application Library** to view the list of available applications.
6. Choose the **GitLab server** application.
    <Message type="tip">
      If you cannot find the GitLab server application on the first page, use the search bar or navigate through the library.
    </Message>
7. Customize the default configuration for GitLab server using [Helm Charts](/tutorials/kubernetes-package-management-helm/). The default configuration looks as follows:
    ```yaml
    certmanager-issuer:
      email: me@example.com
    global:
      hosts:
        domain: gitlab.example.com
    ```

>     Replace the parameters:

Reviewer: ```suggestion
    Replace the following parameters:
```

    - `email`: Your email address.
    - `domain`: Your domain name.
        <Message type="note">
          The GitLab Server application is configured to use a `gitlab` subdomain. If you want your application running at `gitlab.example.com`, enter `example.com`.
        </Message>
8. Enter a name (e.g., `gitlab-server`) and a Kubernetes namespace for your application. If no name is entered, GitLab server will be installed in the default namespace.
9. Click **Deploy Application** to deploy the GitLab server on your Kubernetes cluster.

## Retrieving the GitLab Server root password

1. Access the [Kubernetes Dashboard](/containers/kubernetes/how-to/access-kubernetes-dashboard/) of your cluster and navigate to **Secrets** under the **Config and Storage** section.
2. Click `gitlab-server-gitlab-initial-root-password` to display the application metadata.

> 3. Click the "eye" icon next to `password` to reveal and copy the GitLab server root password.

Reviewer 2: ```suggestion
3. Click the eye icon next to `password` to reveal and copy the GitLab server root password.
```

## Connecting to GitLab Server

> GitLab server uses a LoadBalancer to manage traffic. To check the deployment status, run:

Reviewer: ```suggestion
GitLab server uses a Load Balancer to manage traffic. To check the deployment status, run:
```

```bash
$ kubectl get svc
```

Look for the `gitlab-server-nginx-ingress-controller` service. Once the `EXTERNAL-IP` status changes from `<PENDING>` to an assigned IP, your GitLab server application is up.

> Configure an A-record for the `gitlab` subdomain in your DNS zone, pointing to the Load Balancer IP.

Reviewer 3: Having no domain in hand, I wasn't able to go beyond this point. I trust it goes well beyond here :)

Author: Tried this and checked it with Pablo :) A domain is required to register the runner, otherwise the runner of the server fails to start
